### PR TITLE
changed gitignore handling

### PR DESCRIPTION
**Pull request by mzuenni**
#305

### CODE REVIEW

**Comment by mpsijm** [comment]
Thanks for simplifying the implementation, but I think I still don't like having `.gitignore` files in the root of my problem directory :sweat_smile: I think we need more opinions here :stuck_out_tongue:

**Comment by mzuenni** [comment]
it should be easy to change now ^^ (as long as the `gitignore` stays in the problem folder). I just hope that i found all places that needed a change

**Comment by mpsijm** [comment]
> it should be easy to change now ^^ (as long as the `gitignore` stays in the problem folder).

But it's not easy to change when I don't have a `.gitignore` file in my problem folders, and BAPCtools keeps regenerating it :sweat_smile:

**Comment by mzuenni** [comment]
I meant it should be easy to change the code if we decide for the self ignoring gitignore in `/data`.

File: bin/generate.py [hunk 1]
```diff
@@ -1699,60 +1700,24 @@ def generate_dir_unlisted(d):
...
         if gitignorefile.is_file():
-            text = gitignorefile.read_text()
-            if not text.startswith('# GENERATED BY BAPCtools'):
-                warn('Not overwriting existing data/.gitignore file.')
-                return
+            # if there is any rule fore data/ we expect that the user knows
+            # what he does.
+            if gitignorefile.read_text().find('data/') == -1:
+                with gitignorefile.open("a") as f:
+                    f.write(content)
```
Comment (RagnarGrootKoerkamp) [inline bin/generate.py:1715]: Maybe write a preceding newline to not create bugs when `f` doesn't end in a newline.

File: skel/problem/.gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#GENERATED BY BAPCtools
```
Comment (RagnarGrootKoerkamp) [inline skel/problem/.gitignore:1]: I think only writing the .gitignore once `bt generate` has been invoked is preferable. That way we don't ignore things if the user doesn't use `bt generate`.

**Comment by RagnarGrootKoerkamp** [comment]
I also like putting this in the problem .gitignore. Indeed it's a bit redundant but I agree it's more a problem than a contest setting, and anyway these files are automatically generated so there is little overhead.

So this settles A1 and B2 from #305. That only leaves the cleanup of unknown-file handling (C).

**Review by RagnarGrootKoerkamp — APPROVED** [review]

**Pull request merged**